deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c

```diff
@@ -71,25 +136,116 @@ GrB_Info GxB_MatrixTupleIter_jump_to_row
 ) {
 	GB_WHERE1("GxB_MatrixTupleIter_jump_to_row (iter, rowIdx)") ;
 	GB_RETURN_IF_NULL(iter) ;
+	if(iter->iterators_size == 0) return GrB_INVALID_VALUE ;
 
-	// Deplete iterator, should caller ignore returned error.
-	_EmptyIterator(iter) ;
+	for(size_t i = 0; i < iter->iterators_size; i++) {
+		_GxB_MatrixTupleIter *iter_cur = &iter->iterators[i] ;
 
-	if(rowIdx < 0 || rowIdx >= iter->nrows) {
-		GB_ERROR(GrB_INVALID_INDEX,
-				 "Row index " GBu " out of range; must be < " GBu,
-				 rowIdx, iter->nrows) ;
-	}
+		// Deplete iterator, should caller ignore returned error.
+		_EmptyIterator(iter_cur) ;
 
-	GrB_Matrix_nvals(&(iter->nvals), iter->A) ;
+		if(rowIdx < 0 || rowIdx >= iter_cur->nrows) {
+			GB_ERROR(GrB_INVALID_INDEX,
+					"Row index " GBu " out of range ; must be < " GBu,
+					rowIdx, iter_cur->nrows) ;
+		}
 
-	iter->p        =  0                   ;
-	iter->nnz_idx  =  iter->A->p[rowIdx]  ;
-	iter->row_idx  =  rowIdx              ;
+		GrB_Index _rowIdx = rowIdx ;
+		if(iter_cur->sparsity_type == GxB_HYPERSPARSE) {
+			GrB_Index rowIdx_sparse ;
+			if(!_find_row_index_in_Ah(iter_cur, rowIdx, &rowIdx_sparse)) {
+				GB_ERROR (GrB_INVALID_INDEX,
+					"Row index " GBu " doesn't exist in the hypersparse matrix, row might be empty",
+					rowIdx ) ;
+			}
+			_rowIdx = rowIdx_sparse ;
+			iter_cur->sparse_row_idx = rowIdx_sparse ;
+			iter_cur->h_size = iter_cur->A->nvec ;
+		}
+		GrB_Matrix_nvals(&(iter_cur->nvals), iter_cur->A) ;
+
+		iter_cur->p        =  0                   ;
+		iter_cur->nnz_idx  =  iter_cur->A->p[_rowIdx]  ;
+		iter_cur->row_idx  =  rowIdx              ;
+	}
 
 	return (GrB_SUCCESS) ;
 }
 
+// finds the start row index in Ah for HYPERSPARSE matrix, 
+// returns true if found else false.
+static inline bool _find_start_row_index_in_Ah
+(
+	_GxB_MatrixTupleIter *iter,
+	GrB_Index rowIdx,
+	GrB_Index *result
+) {
+	GB_WHERE1("_find_start_row_index_in_Ah (iter, rowIdx, result)") ;
+	if(!iter->h_size) return false ;
+	GrB_Matrix A = iter->A ;
+	bool found ;
+	GrB_Index l = 0, r = iter->h_size - 1 ;
+
+	GB_BINARY_SEARCH(rowIdx, A->h, l, r, found) ;
+	if(found) {
+		*result = l ;
+		return found ;
+	} else if(A->h[l] > rowIdx) {
+		// rowIdx not found, look for the minimal row index which is greater than rowIdx
+		// this can be located in h[l] or h[l+1]
+		*result = l ;
+		return true ;
+	} else if(l + 1 < iter->h_size) {
+		if(A->h[l + 1] <= rowIdx) {
+			GB_ERROR(GrB_INVALID_VALUE,
+				"row index " GBu " must be > than rowIdx" GBu,
+				A->h[l + 1], rowIdx) ;
+		}
+		*result = l+1 ;
+		return true ;
+	}
+
+	// reaching here means no start row index found
+	return false ;
+}
+
+// finds the end row index in Ah for HYPERSPARSE matrix,
+// returns true if found else false.
+static inline bool _find_end_row_index_in_Ah
```

Comment:
Please rename to indicate we're searching for an index that is the maximal value that satisfies `<= rowIdx`